### PR TITLE
CORE-18839 - Kafka client upgrade

### PR DESCRIPTION
**Pull request by driessamyn**
Move Kafka Client to version catalog and update to version that doesn't need snappy override.

### CODE REVIEW

File: components/membership/membership-persistence-service-impl/build.gradle [hunk 1]
```diff
@@ -52,7 +52,7 @@ dependencies {
     integrationTestImplementation project(':libs:layered-property-map')
     integrationTestImplementation project(':libs:messaging:topic-admin')
     integrationTestImplementation project(':testing:message-patterns')
-    integrationTestImplementation "org.apache.servicemix.bundles:org.apache.servicemix.bundles.kafka-clients:$kafkaClientVersion"
+    integrationTestImplementation libs.kafka.client
```
Comment (yift-r3) [inline components/membership/membership-persistence-service-impl/build.gradle:55]: Do we need this?

**Review by yift-r3 — APPROVED** [review]

File: components/membership/membership-persistence-service-impl/build.gradle (hunk 1 above)
Comment (driessamyn) [reply components/membership/membership-persistence-service-impl/build.gradle:55]: I don't know, but it  wasn't added in this PR, so would rather not take out as part of this change as I'm not familiar with these tests.

**Review by LWogan — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by wzur-r3 — APPROVED** [review]
LGTM for Gradle files

**Pull request merged**